mm2src/coins/utxo_signer/src/with_key_pair.rs

```diff
@@ -77,20 +75,30 @@ pub fn p2pk_spend(
     fork_id: u32,
 ) -> UtxoSignWithKeyPairResult<TransactionInput> {
     let unsigned_input = get_input(signer, input_index)?;
-
-    let script = Builder::build_p2pk(key_pair.public());
-    if script != unsigned_input.prev_script {
+    // P2PK UTXOs can have either compressed or uncompressed public keys in the scriptPubkey.
+    // We need to check that one of them matches the prev_script of the input being spent.
+    let pubkey = key_pair.public().to_secp256k1_pubkey().map_err(|e| {
+        UtxoSignWithKeyPairError::InternalError(format!("Couldn't get secp256k1 pubkey from keypair: {}", e))
+    })?;
+    // Build the scriptPubKey for both compressed and uncompressed public keys.
+    let possible_script_pubkeys = vec![
+        Builder::build_p2pk(&keys::Public::Compressed(pubkey.serialize().into())),
+        Builder::build_p2pk(&keys::Public::Normal(pubkey.serialize_uncompressed().into())),
+    ];
+    // Check that one of the scriptPubkeys matches the prev_script of the input being spent.
+    if !possible_script_pubkeys.contains(&unsigned_input.prev_script) {
         return MmError::err(UtxoSignWithKeyPairError::MismatchScript {
             script_type: "P2PK".to_owned(),
-            script,
+            // Safe unwrapping since the array has exactly 2 elements.
+            script: possible_script_pubkeys.get(0).unwrap().clone(),
```

Comment:
Since it' safe, you can do it in the `t[0]` way or use [get_unchecked ](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get_unchecked)instead.

Reply:
indexing complains:
![image](https://github.com/user-attachments/assets/47b879d4-edfb-420d-ab4c-93ab640fce17)

get_unchecked is unsafe. i dont wanna label the block as unsafe for such a tiny thing.

i feel there is some way more idiomatic than `get(0).unwrap()` but can't find it :(.

Reply:
[d3d4ab2](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2410/commits/d3d4ab2e4d07e7f18b0f2e4b8e2350cd496eab71)

found this, looks better.

Reply:
> i feel there is some way more idiomatic than get(0).unwrap() but can't find it :(.

I think it is more of trying to prevent bound checks...using get will add unnecessary bound checks even tho we can guarantee that we will have exactly two items!